src/wp-admin/includes/image.php

```diff
@@ -647,18 +647,27 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
  * @since 2.5.0
  *
  * @param string $str
- * @return int|float
+ * @return int|float Returns 0 on failure.
  */
 function wp_exif_frac2dec( $str ) {
-	if ( false === strpos( $str, '/' ) ) {
-		return $str;
+	// Fractions must contain a single `/`.
+	if ( substr_count( $str, '/' ) !== 1 ) {
+		return 0;
 	}
 
 	list( $numerator, $denominator ) = explode( '/', $str );
-	if ( ! empty( $denominator ) ) {
-		return $numerator / $denominator;
+
+	// Both the numerator and the denominator must be numbers.
+	if ( ! is_numeric( $numerator ) || ! is_numeric( $denominator ) ) {
+		return 0;
+	}
+
+	// The denominator must not be empty (or 0).
+	if ( $denominator === 0 ) {
```

Comment:
testing in PHP8, I see `list( $numerator, $denominator ) = explode( '/', $str );` returning an empty denominator (not 0 but an empty string) for the test case '100/0'. Not certain why, but feels safer to use empty test here anyway.

Reply:
I cannot reproduce what you are saying: https://3v4l.org/llsu0

Reply:
ooh, that eval tool is cool!

I see what was happening now, the denominator is "0" in this case: that is a string of the number 0. 

In this case `0 ===` returns false, and still messes up the division. `empty()` returns true and original value is returned.

Reply:
since i restored is_numeric, "0" wouldn't make it here, will try restoring